src/catalog/dependency_manager.cpp

```diff
@@ -128,26 +129,113 @@ void DependencyManager::EraseObject(CatalogEntry &object) {
 	EraseObjectInternal(object);
 }
 
+catalog_entry_set_t &DependencyManager::GetEntriesThatObjectDependsOn(CatalogEntry &object) {
+	return dependencies_map.at(object);
+}
+
+dependency_set_t &DependencyManager::GetEntriesThatDependOnObject(CatalogEntry &object) {
+	return dependents_map.at(object);
+}
+
 void DependencyManager::EraseObjectInternal(CatalogEntry &object) {
 	if (dependents_map.find(object) == dependents_map.end()) {
 		// dependencies already removed
 		return;
 	}
 	D_ASSERT(dependents_map.find(object) != dependents_map.end());
 	D_ASSERT(dependencies_map.find(object) != dependencies_map.end());
-	// now for each of the dependencies, erase the entries from the dependents_map
-	for (auto &dependency : dependencies_map[object]) {
-		auto entry = dependents_map.find(dependency);
-		if (entry != dependents_map.end()) {
-			D_ASSERT(entry->second.find(object) != entry->second.end());
-			entry->second.erase(object);
+
+	auto &entries = GetEntriesThatObjectDependsOn(object);
+	for (auto &other : entries) {
+		// For every entry that 'object' is dependent on, clean up this connection
+		auto dependencies_entry = dependents_map.find(other);
+		if (dependencies_entry == dependents_map.end()) {
+			continue;
 		}
+
+		auto &dependencies = dependencies_entry->second;
+		auto dependent_entry = dependencies.find(object);
+		D_ASSERT(dependent_entry != dependencies.end());
+
+		// Remove the dependency of 'object' on 'other'
+		dependencies.erase(dependent_entry);
 	}
 	// erase the dependents and dependencies for this object
 	dependents_map.erase(object);
 	dependencies_map.erase(object);
 }
 
+void DependencyManager::PrintDependencyMap() {
```

Comment:
These are for debugging purposes.
What would be the best way to keep them around so they can be useful in the future without causing unnecessary friction?

Reply:
Would it be desirable to expose this (in some form) to SQL?

Idea is that then it can be tested / and being present there for debug purposes.

Reply:
That's not a bad idea, maybe we could make two table functions that output a column for every catalog entry, containing a VARCHAR[] containing the names of the dependents/dependencies

Reply:
Btw I just found that we have `duckdb_dependencies()` table function, but it seems kind of lacking